dali/pipeline/operator/builtin/external_source.cc

```diff
@@ -18,11 +18,12 @@
 
 namespace dali {
 
-template <>
+template<>
 void ExternalSource<CPUBackend>::RunImpl(Workspace &ws) {
   auto &output = ws.Output<CPUBackend>(0);
   auto &thread_pool = ws.GetThreadPool();
   ForwardCurrentData(output, data_id_, thread_pool);
+  SetDepletedOperatorTrace(ws, !(repeats_last_ || HasDataInQueue()));
```

Comment:
```suggestion
  SetDepletedOperatorTrace(ws, HasDataInQueue());
```
1. If no data was _ever_ fed to an `ExternalSource`, it will be "depleted".
2. The pipeline takes care of refeeding the ExternalSource - the operator itself doesn't even need to be aware of the value of `repeat_last`.

Reply:
Sorry, I missed that we've just removed the (possibly only) entry from the queue.